helper/logging/transport.go

```diff
@@ -0,0 +1,53 @@
+package logging
+
+import (
+	"log"
+	"net/http"
+	"net/http/httputil"
+)
+
+type transport struct {
+	name      string
+	transport http.RoundTripper
+}
+
+func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if IsDebugOrHigher() {
+		reqData, err := httputil.DumpRequest(req, true)
```

Comment:
Since this is an outgoing request we should probably use DumpRequestOut (I'm not sure what the practical difference is but the godoc seems to point to using this one instead)